src/SourceBuild/content/eng/sdkArchiveDiff.targets

```diff
@@ -0,0 +1,64 @@
+<Project>
+
+  <UsingTask AssemblyFile="$(SdkArchiveDiffTasksAssembly)" TaskName="GetValidArchiveItems" />
+  <UsingTask AssemblyFile="$(SdkArchiveDiffTasksAssembly)" TaskName="GetClosestOfficialSdk" />
+  <UsingTask AssemblyFile="$(SdkArchiveDiffTasksAssembly)" TaskName="FindArchiveDiffs" />
+
+  <Target Name="ReportSdkArchiveDiffs"
+          AfterTargets="Build"
+          DependsOnTargets="DetermineSourceBuiltSdkVersion">
+
+      <Message Text="Comparing built SDK against closest official build"
+               Importance="High"/>
+
+      <GetValidArchiveItems ArchiveItems="@(SdkTarballItem)"
+                            ArchiveName="dotnet-sdk">
+        <Output TaskParameter="ValidArchiveItems"
+                ItemName="_BuiltSdkArchivePath"/>
+      </GetValidArchiveItems>
+
+      <!-- There should only be 1 SDK archive -->
+      <Error Text="Multiple valid dotnet-sdk archives found."
+             Condition="'@(_BuiltSdkArchivePath->Count())' != '1'" />
+
+      <GetClosestOfficialSdk BuiltArchivePath="@(_BuiltSdkArchivePath)">
+        <Output TaskParameter="ClosestOfficialArchivePath"
+                PropertyName="_ClosestOfficialSdkPath" />
+      </GetClosestOfficialSdk>
+
+      <FindArchiveDiffs BaselineArchive="@(_BuiltSdkArchivePath)"
+                        TestArchive="$(_ClosestOfficialSdkPath)">
+        <Output TaskParameter="ContentDifferences"
+                ItemName="_ContentDifferences" />
+      </FindArchiveDiffs>
+
+      <ItemGroup>
+        <_changedFiles Include="@(_ContentDifferences)" Condition="'%(_contentDifferences.Kind)' != 'Unchanged'" />
+        <_sdkFilesDiff Include="@(_ContentDifferences)" Condition="'%(_contentDifferences.Kind)' == 'Added'" >
+          <DiffIndicator>+</DiffIndicator>
+        </_sdkFilesDiff>
+        <_sdkFilesDiff Include="@(_ContentDifferences)" Condition="'%(_contentDifferences.Kind)' == 'Removed'" >
+          <DiffIndicator>-</DiffIndicator>
+        </_sdkFilesDiff>
+        <_sdkFilesDiff Include="@(_ContentDifferences)" Condition="'%(_contentDifferences.Kind)' == 'Unchanged'" >
+          <DiffIndicator> </DiffIndicator>
+        </_sdkFilesDiff>
+      </ItemGroup>
+
+      <PropertyGroup>
+        <SdkArchiveDiffsReport>$(ArtifactsLogDir)SdkArchiveContent.diff</SdkArchiveDiffsReport>
```

Comment:
instead of .diff what if we generate xunit/mstest xml reports instead? those could be picked up by the AzDO test tab so we don't have to build custom UI.